scipy/optimize/_differentialevolution.py

```diff
@@ -742,11 +757,20 @@ def maplike_for_vectorized_func(func, x):
         else:
             self.integrality = False
 
+        # check for equal bounds
+        eb = self.limits[0] == self.limits[1]
```

Comment:
If `eb` were defined earlier, it seems it could have been used and avoided the `~np.isfinite(self.__recip_scale_arg2)`

Reply:
point taken. This would then be

```
self.__recip_scale_arg2[eb] = 0
```